synapse/config/server.py

```diff
@@ -521,6 +529,12 @@ class LimitRemoteRoomsConfig(object):
             "request_token_inhibit_3pid_errors", False,
         )
 
+        # Whether new, locally-created rooms should have encryption enabled
+        self.encryption_enabled_by_default_for_room_type = config.get(
+            "encryption_enabled_by_default_for_room_type",
+            RoomDefaultEncryptionTypes.OFF,
+        )
```

Comment:
The stuff at https://github.com/matrix-org/synapse/pull/7639/files#diff-720b6a295d1525b1f9aabce55ba4cfe0R73 feels a bit cumbersome, and I feel like we could improve that and also do some validation here by preprocessing the config setting a bit, by creating a `self.encryption_enabled_by_default_for_room_presets` here?

Reply:
I'm not sure I quite understand your point. Do you mean we would have a boolean in the constructor titled `self.encryption_enabled_by_default_for_room_presets` and later when a room is created, check this boolean along with the desired preset?

I do agree that this dict has the potential to grow massive and horrible.

Reply:
I'm not sure we could move it out of the constructor with what I'm envisaging. Just that the setup code would be simpler.

For clarity, I am imagining that here you would build a set (probably called `encryption_enabled_by_default_for_room_presets`), which would contain zero or more of `public_chat`, `private_chat`, `trusted_private_chat`; and that you would then do a lookup in that set when setting up `_presets_dict`.